guides/hack/08-generic-types-and-functions/20-reified-generics.md

```diff
@@ -0,0 +1,187 @@
+# Reified Generics
+
+## Introduction
+
+Generics are currently implemented in HHVM through erasure, in which the runtime drops all information about generics. This means that generics are not available at runtime. Although the typechecker is able to use the generic types for static typechecking, we are unable to enforce generic types at runtime.
+
+The goal of opt-in reified generics to bridge the gap between generics and runtime availability while keeping erasure available to maintain performance when reification is not needed. To mark a generic as reified, simply add the `reify` keyword at the declaration site.
+
+## Parameter and return type verification
+
+Prior to reified generics, if the type hint for a parameter or return value had generics, HHVM would only check whether the input is of the outermost type (whether `$c` is a `C` in the next example). With reified generics, HHVM will also check and enforce the reified type parameters.
+
+```Hack
```

Comment:
- pseudomain example :'( Please use `<<__EntryPoint>>`
- please split out to .php or .hack files and add .hhvm.expect and .typechecker.expect files (or expectf/expectre files - we use the HHVM test runner)
- instead of commetijng with expected result, please change to output something indicating success (so that the .hhvm.expect file is meaningful)

This should go in a '*examples/' subdirectory, and be included with `@@ foo-examples/foo.hack @@` (or .php) - see the other markdown files here for examples.

This makes sure that our examples actually work, and continue to work on new releases of HHVM and the typechecker